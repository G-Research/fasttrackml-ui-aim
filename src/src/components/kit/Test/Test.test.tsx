import { render } from '@testing-library/react';

import Test from './Test';

const Title1 = 'Test Title 1';
const Title2 = 'Test Title 2';

describe('<Test /> - ', () => {
  test('Renders without crashing', () => {
    const { asFragment } = render(<Test title={Title1} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('Renders right content', () => {
    const { getByText } = render(<Test title={Title2} />);

    getByText(Title2);
  });

  test('Prop change works properly', () => {
    const { getByText, rerender } = render(<Test title={Title2} />);
    getByText(Title2);
    rerender(<Test title={Title1} />);
    getByText(Title1);
  });
});
