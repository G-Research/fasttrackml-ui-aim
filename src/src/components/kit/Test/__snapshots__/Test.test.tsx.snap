// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test /> -  Renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="Test"
  >
    <span
      class="Text Text__size_12 Text__weight_500 Text__color_primary"
    >
      Test Title 1
    </span>
  </div>
</DocumentFragment>
`;
